src/util/dates.js

```diff
@@ -623,6 +623,26 @@ export const formatDateStringToUTC = dateString => {
   return moment.utc(dateString).toDate();
 };
 
+/**
+ * Format given date string ('YYYY-MM-DD') to a full date string in
+ * the given time zone.
+ *
+ * This is used in search when filtering by time-based availability.
+ *
+ * Example:
+ * ('2020-04-15', 'Etc/UTC') => '2020-04-15T00:00:00.000Z'
+ * ('2020-04-15', 'Europe/Helsinki') => '2020-04-14T21:00:00.000Z'
+ *
+ * @param {String} dateString in 'YYYY-MM-DD' format
+ * @param {String} tzId time zone id, see:
+ *   https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+ *
+ * @returns {String} string in '0000-00-00T00:00:00.000Z' format
+ */
+export const formatDateStringToTz = (dateString, tzId) => {
+  return moment.tz(dateString, tzId).toDate();
```

Comment:
Other functions in this util file use 'timeZone' instead of 'tzId'. It's probably better to be consistent.